### ISSUE TITLE
Query failures are not detected correctly

### ISSUE THREAD

**Issue by metaodi** [ruflin/Elastica#387]
Today I tried to setup [sorting with Elastica/ElasticSearch](http://www.elasticsearch.org/guide/reference/api/search/sort/). While trying to setup I used an analysed string property for sorting, which simply led to an empty result set. Elastica did **not** indicate an error or threw an Exception.

Example query where "name" is analysed:
```json
{
  "sort": {
    "name": {
      "order": "asc"
    }
  },
  "query": {
    "match_all": {}
  }
}
```

Then the return value of ElasticSearch is a JSON with an error:
```json
{
error: SearchPhaseExecutionException[Failed to execute phase [query], total failure; shardFailures {[3ChPwM16TfmiF8U-fSyzNw][de][1]: QueryPhaseExecutionException[[de][1]: query[ConstantScore(NotDeleted(cache(_type:product)))],from[0],size[10],sort[<custom:"name": org.elasticsearch.index.field.data.strings.StringFieldDataType$1@63565c27>]: Query Failed [Failed to execute main query]]; nested: IOException[Can't sort on string types with more than one value per doc, or more than one token per field]; }
...
status: 500
}
```

I tried the same with an "sub-object".  In this case "labels" contains an array of "label" object, which each have "analysed" name properties.

Request:
```json
{
  "sort": {
    "labels.name": {
      "order": "asc"
    }
  },
  "query": {
    "match_all": {}
  }
}
```

I didn't get an error object, but a response from ElasticSearch: 
```json
{
    took: 12
    timed_out: false
    _shards: {
        total: 5
        successful: 3
        failed: 2
        failures: [{
                index: de
                shard: 1
                status: 500
                reason: QueryPhaseExecutionException[[de][1]: query[ConstantScore(NotDeleted(cache(_type: product)))], from[0], size[10], sort[ < custom: "labels.name": org.elasticsearch.index.field.data.strings.StringFieldDataType$1@ 130e709b > ]: Query Failed[Failed to execute main query]];
                nested: IOException[Can 't sort on string types with more than one value per doc, or more than one token per field]; 
}
{
index: de
shard: 0
status: 500
reason: QueryPhaseExecutionException[[de][0]: query[ConstantScore(NotDeleted(cache(_type:product)))],from[0],size[10],sort[<custom:"labels.name": org.elasticsearch.index.field.data.strings.StringFieldDataType$1@430fc856>]: Query Failed [Failed to execute main query]]; nested: IOException[Can'
                t sort on string types with more than one value per doc, or more than one token per field];
            }
        ]
    }
    hits: {
        total: 5
        max_score: null
        hits: [{
                _index: de
                _type: product
                _id: 703031700000
                _score: null
                _source: {
                    id: 703031700000
    ...
```

As you can see, the response contains successful **and** failed entries. This case is not handled correctly by Elastica. There is no error message or exception that indicates this. The return value a result set containing only the successful entries.These are those, where `labels` is an empty array (i.e. the sorting does not fail on those).

Unlike the full "error" that is indicated by the error property, this case is more subtle as it is indicated by the `failure` property of the `_shards`.

**UPDATE:** I discussed this issue with @Tobion and cleared some stuff up.

**Comment by im-denisenko** [comment]
@ruflin Isn't it fixed by #553? Can we close this?

**Comment by ruflin** [comment]
Yep
